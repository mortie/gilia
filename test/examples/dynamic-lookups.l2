# Lookup array values dynamically
idx := 0
arr := [100 20 30]
print arr.(idx)

# More complex expressions in the subscript operation
arr.(+ idx 1) = 50
print arr

# Lookup into a temporary
print [10 20 30].(+ 1 1)

# Lookup into a namespace by atom
obj := {}
obj.('hello) = "what's up"
print obj.hello

# More complicated expression again
get-ident := {'foo}
obj.(get-ident()) = 100
print obj.foo
