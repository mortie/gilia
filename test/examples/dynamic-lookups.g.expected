100
[100 50 30]
30
what's up
100
